api/src/wfl/module/covid.clj

```diff
@@ -958,17 +959,23 @@
 
 (defn verify-terra-sink!
   "Verify that the WFL has access the `workspace`."
-  [{:keys [workspace entityType skipValidation] :as sink}]
-  (if skipValidation
-    sink
-    (do (when-not (do-or-nil (firecloud/workspace workspace))
-          (throw (UserException. "Cannot access workspace"
-                                 (util/make-map workspace))))
-        (let [types (->> workspace firecloud/list-entity-types keys set)]
-          (when-not (-> entityType keyword types)
-            (throw (UserException. "Entity not found"
-                                   (util/make-map entityType types workspace)))))
-        sink)))
+  [{:keys [entityType fromOutputs skipValidation workspace] :as sink}]
+  (when-not skipValidation
+    (when-not (do-or-nil (firecloud/workspace workspace))
+      (throw (UserException. "Cannot access workspace"
+                             (util/make-map workspace))))
+    (let [entity-type  (keyword entityType)
+          entity-types (firecloud/list-entity-types workspace)
+          types        (-> entity-types keys set)]
+      (when-not (types entity-type)
+        (throw (UserException. "Entity not found"
+                               (util/make-map entityType types workspace))))
+      (let [attributes    (get-in entity-types [entity-type :attributeNames])
+            [missing _ _] (data/diff (set (vals fromOutputs)) (set attributes))]
+        (when (seq missing)
+          (throw (UserException. "Attributes missing for these outputs"
+                                 (util/make-map missing)))))))
```

Comment:
Might be useful to know what the actual attributes are too, not just the missing ones. Even better sorted lexicographically.

Reply:
Typically there will be lots of attributes but just a few missing ones,
but added and sorted.  We can revisit if `:missing` gets swamped.